hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackAware.java

```diff
@@ -137,10 +137,6 @@ public void chooseNodeWithNoExcludedNodes() throws SCMException {
         datanodeDetails.get(2)));
     Assert.assertFalse(cluster.isSameParent(datanodeDetails.get(1),
         datanodeDetails.get(2)));
-    Assert.assertFalse(cluster.isSameParent(datanodeDetails.get(0),
-        datanodeDetails.get(3)));
-    Assert.assertFalse(cluster.isSameParent(datanodeDetails.get(2),
-        datanodeDetails.get(3)));
```

Comment:
I guess you are removing this assertion because it makes the test flaky (node 2 and 3 may or may not be in the same rack, the only restriction is that they are in different one than nodes 0 and 1).  If so, please also remove similar assertion from `testFallback`:

https://github.com/apache/hadoop/blob/f9266ccc4d592681cc8227a147799d3a08623493/hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackAware.java#L200-L201

Reply:
Thanks for the comments. Will remove last two assertions in testFallback.